sdk/core/Azure.Core.Experimental/src/DynamicData.ArrayEnumerator.cs

```diff
@@ -8,13 +8,13 @@
 
 namespace Azure.Core.Dynamic
 {
-    public partial class DynamicJson
+    public partial class DynamicData
     {
         /// <summary>
         /// An enumerable and enumerator for the contents of a mutable JSON array.
         /// </summary>
         [DebuggerDisplay("{Current,nq}")]
-        public struct ArrayEnumerator : IEnumerable<DynamicJson>, IEnumerator<DynamicJson>
+        public struct ArrayEnumerator : IEnumerable<DynamicData>, IEnumerator<DynamicData>
         {
             private MutableJsonElement.ArrayEnumerator _enumerator;
             private readonly DynamicJsonOptions _options;
```

Comment:
It's a little awkward to hold a JSON-specific options here, but given this is all internal, I think it leaves space for extensibility.

Reply:
Though, it is `public`. Should probably be marked `private`. Enumerables in the .NET BCL almost always are. It only matters they implement `IEnumerable<T>` anyway.

Reply:
It is modeled after JsonElement's public [ArrayEnumerator](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.arrayenumerator?view=net-8.0), but you make a good point.  If everything else works, we can always make it public later if there is a reason.  Let me look at hiding it.  Thanks, @heaths!

Reply:
Updated per your feedback, @heaths - thanks!